src/main/java/com/thebois/models/world/structures/Stockpile.java

```diff
@@ -0,0 +1,47 @@
+package com.thebois.models.world.structures;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+import com.thebois.models.Position;
+import com.thebois.models.inventory.IInventory;
+import com.thebois.models.inventory.items.ItemType;
+
+/**
+ * A structure used for storing items.
+ */
+public class Stockpile extends AbstractStructure {
+
+    private final IInventory inventory;
+
+    Stockpile(final Position position, final IInventory inventory) {
+        super(position, StructureType.STOCKPILE, neededItems());
+        this.inventory = inventory;
+    }
+
+    private static Map<ItemType, Integer> neededItems() {
+        final Map<ItemType, Integer> neededItems = new EnumMap<>(ItemType.class);
+
+        final int numberOfItems = 4;
+
+        neededItems.put(ItemType.LOG, numberOfItems);
+        neededItems.put(ItemType.ROCK, numberOfItems);
+
+        return neededItems;
+    }
+
+    @Override
+    public IStructure deepClone() {
+        return new Stockpile(getPosition(), inventory);
+    }
+
+    @Override
+    public float getCost() {
+        return Float.MAX_VALUE;
+    }
+
+    public IInventory getInventory() {
+        return inventory;
```

Comment:
We might need to change the `ArrayList` used in Inventory to a `ConcurrentLinkedQueue` since all pawns accessing this will probably throw a `ConcurrentModificationException`, but I guess we will find out later.

Reply:
We're running everything on one thread, so why would it throw anything? And that's a queue not a list, we shouldn't keep items in a queue, their order doesn't matter.